pageserver/src/remote_storage/storage_sync/download.rs

```diff
@@ -149,6 +158,19 @@ where
                             "Failed to download a layer from storage path '{layer_storage_path:?}'"
                         )
                     })?;
+                destination_file.sync_data().await?;
+
+                // fsync parent dir
+                let parent_path = temp_file_path
+                    .parent()
+                    .expect("no parent dir for layer file");
+                let parent_dir = fs::File::open(parent_path).await?;
+                parent_dir.sync_all().await?;
+
+                fail::fail_point!("remote-storage-download-pre-rename", |_| {
+                    bail!("failpoint triggered")
```

Comment:
NIT: could be more specific on what point exactly was triggered, so somebody could grep it easily.